cmd/pulumictl/convert-version/cli.go

```diff
@@ -0,0 +1,66 @@
+package convertVersion
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/pulumi/pulumictl/pkg/gitversion"
+	"github.com/spf13/cobra"
+	"github.com/spf13/viper"
+)
+
+var (
+	language string
+	version  string
+)
+
+func Command() *cobra.Command {
+	command := &cobra.Command{
+		Use:   "convert-version",
+		Short: "Convert versions",
+		Long:  "Convert a generic version into a language specific version",
+		Args:  cobra.MaximumNArgs(1),
+
+		RunE: func(cmd *cobra.Command, args []string) error {
+			if err := cmd.ParseFlags(args); err != nil {
+				return err
+			}
+
+			language = viper.GetString("language")
+			version = viper.GetString("version")
+
+			versions, err := gitversion.GetLanguageOptionsFromVersion(version)
+
+			if err != nil {
+				return fmt.Errorf("error calculating version: %w", err)
+			}
+
+			// FIXME: We could get the values here from the struct fields?
+			switch strings.ToLower(language) {
+			case "generic":
+				fmt.Println(versions.SemVer)
+			case "python":
+				fmt.Println(versions.Python)
+			case "javascript":
+				fmt.Println(versions.JavaScript)
+			case "dotnet":
+				fmt.Println(versions.DotNet)
+			default:
```

Comment:
Should "generic" be the default case, or is it better to explicitly error here?

Reply:
I don't really understand cobra - for some reason if it's not specified it comes through as "generic" ... but I'm not giving it that default. Can only assume there's some kind of global state going on internally that means the default gets set by one of the other commands 🤦